rkt/image/httpops.go

```diff
@@ -91,19 +91,19 @@ func (o *httpOps) DownloadImageWithETag(u *url.URL, etag string) (readSeekCloser
 	if err != nil {
 		return nil, nil, err
 	}
-	defer func() { maybeClose(aciFile) }()
```

Comment:
You are closing this if download fails, but you still leak it if `cd.UseCached` is `true`…